azure/Guidelines.md

```diff
@@ -0,0 +1,865 @@
+If-Modified-Since   | Request    | Sun, 06 Nov 1994 08:49:37 GMT (see [Conditional Requests](#Conditional-Requests))
+If-Unmodified-Since | Request    | Sun, 06 Nov 1994 08:49:37 GMT (see [Conditional Requests](#Conditional-Requests))
+date                | Both       | Sun, 06 Nov 1994 08:49:37 GMT (see [RFC7231, Section 7.1.1.2](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.2))
+*content-type*      | Both       | application/merge-patch+json
+*content-length*    | Both       | 1024
+*x-ms-request-id*   | Response   | [see Customer Support](http://TODO:link-goes-here)
+etag                | Response   | "67ab43" see [Conditional Requests](#Conditional-Requests)
+last-modified       | Response   | Sun, 06 Nov 1994 08:49:37 GMT
+*x-ms-error-code*   | Response   | see [Handling Errors](#Handling-Errors)
+retry-after         | Response   | 180 (see [RFC 7231, Section 7.1.3](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.3))
+
+> :white_check_mark: **DO** support all headers shown in *italics*
+> 
+> :white_check_mark: **DO** specify headers using kebab-casing
+> 
+> :white_check_mark: **DO** compare request header names using case-insensitivity
+> 
+> :white_check_mark: **DO** compare request header values using case-sensitivity if the header name requires it
+> 
+> :white_check_mark: **DO** accept and return date values in headers using the HTTP Date format as defined in [RFC 7231, Section 7.1.1.1](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.1), e.g. "Sun, 06 Nov 1994 08:49:37 GMT"
+
+> :no_entry: **DO NOT** fail a request that contains an unrecognized header. Headers may be added by API gateways or middleware and this must be tolerated
+> 
+> :no_entry: **DO NOT** use "x-" prefix for custom headers, unless the header already exists in production [[RFC 6648](https://datatracker.ietf.org/doc/html/rfc6648)].
+
+#### Additional References
+* [StackOverflow - Difference between http parameters and http headers](https://stackoverflow.com/questions/40492782)
+* [Standard HTTP Headers](https://httpwg.org/specs/rfc7231.html#header.field.registration)
+* [Why isn't HTTP PUT allowed to do partial updates in a REST API?](https://stackoverflow.com/questions/19732423/why-isnt-http-put-allowed-to-do-partial-updates-in-a-rest-api)
+
+### REpresentational State Transfer (REST)
+REST is an architectural style with broad reach that emphasizes scalability, generality, independent deployment, reduced latency via caching, and security. When applying REST to your API, you define your service’s resources as a collections of items. These are typically the nouns you use in the vocabulary of your service. Your service's [URLs](#URLS) determine the hierarchical path developers use to perform CRUD (create, read, update, and delete) operations on resources. Note, it's important to model resource state, not behavior. There are patterns, later in these guidelines, that describe how to invoke behavior on your service.
```

Comment:
Considering the controversy in the past about whether we were really RESTful in the Fielding sense, should we have some remarks about our definition of REST?

Reply:
We could point to the Azure Architecture Center [description on REST](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#what-is-rest) which concludes with the statement:
> In practice, many published web APIs fall somewhere around level 2.

I don't have all the history on this, but I think the content there generally reflects the approach taken in these guidelines.

Reply:
+1